app/src/org/commcare/android/tasks/DataPullTask.java

```diff
@@ -259,6 +259,7 @@ public void reportProgress(int progress) {
                         this.context.sendBroadcast(i);
                         
                         Logger.log(AndroidLogger.TYPE_USER, "User Sync Successful|" + username);
+                        CommCareApplication._().getSession().setCurrentUser(CommCareApplication._().getCurrentUserFromDB(username), password);

```

Comment:
Might be good to use the same db user fetching method that `updateUserSyncToken` uses, instead of using `getCurrentUserFromDB`.
Maybe something like:
```
    private void updateSessionUser(String password) throws SessionUnavailableException {
        SqlStorage<User> storage = CommCareApplication._().getUserStorage("USER", User.class);
        User u = storage.getRecordForValue(User.META_USERNAME, username);
        CommCareApplication._().getSession().setCurrentUser(u, password);
    }
```
I think using `getUserStorage` offers another layer of protection over `getRawStorage`, though I might be mistaken; I don't really understand all the different ways to access the db.